src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -5804,6 +5828,74 @@ public enum EventManifestOptions
         AllowEventSourceOverride = 0x8,
     }
 
+#if FEATURE_EVENTSOURCE_XPLAT 
+
+    internal  class XplatEventLogger : EventListener
+    {
+        public XplatEventLogger() {}
+        
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        public static extern bool IsEventSourceLoggingEnabled();
+
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern void LogEventSource(int eventID, string eventName, string eventSourceName, string payload);
+
+        private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnlyCollection<object> payload, string sep = ", ")
+        {
+
+            if (payloadName == null || payload == null )
+                return String.Empty;
+
+            if (payloadName.Count == 0 || payload.Count == 0)
+                return String.Empty;
+
+            Contract.Assert(payloadName.Count == payload.Count);
+            
+            var sb = StringBuilderCache.Acquire();
+
+            for (int i = 0; i < payloadName.Count; i++)
```

Comment:
Is this actually creating JSON?  This looks to me like it is doing <key>:<value>,<key:value>, etc.  Let's make sure that we're doing valid JSON so that consumers can use a JSON library to parse at the consumer.